Foundation/Bits.hs

```diff
@@ -39,3 +45,27 @@ alignRoundDown :: Int -- ^ Number to Align
                -> Int -- ^ Alignment (power of 2)
                -> Int
 alignRoundDown m alignment = m .&. complement (alignment-1)
+
+-- | perform conversion from Host to Network endianness
+htons :: Word16 -> Word16
+htons w = case endianness of
+    LittleEndian -> byteSwap16 w
+    BigEndian    -> w
```

Comment:
I would really like to get away from host/network terminology (which only really apply to ipv4 in our case, as ipv6 do something different). I would much prefer to just have Little Endian and Big Endian memory abstraction